src/CommunityToolkit.Aspire.Hosting.LavinMQ/LavinMQContainerResource.cs

```diff
@@ -0,0 +1,34 @@
+using CommunityToolkit.Aspire.Hosting.LavinMQ;
+
+namespace Aspire.Hosting.ApplicationModel;
+
+/// <summary>
+/// Represents a container resource for LavinMQ with configurable authentication parameters
+/// and connection endpoint details, implementing the IResourceWithConnectionString interface.
+/// </summary>
+/// <param name="name">The name of the LavinMQ resource instance.</param>
+public class LavinMQContainerResource(string name)
+    : ContainerResource(name), IResourceWithConnectionString
+{
+    private const string DefaultUserName = "guest";
+    private const string DefaultPassword = "guest";
+    internal const string PrimaryEndpointName = "amqp";
+    internal const string ManagementEndpointName = "management";
+    internal const string PrimaryEndpointSchema = "amqp";
+    internal const string ManagementEndpointSchema = "http";
+    internal const int DefaultAmqpPort = 5672;
+    internal const int DefaultManagementPort = 15672;
+
+    private EndpointReference? _primaryEndpoint;
+    private EndpointReference PrimaryEndpoint => _primaryEndpoint ??= new EndpointReference(this, PrimaryEndpointName);
+
+    private static ReferenceExpression UserNameReference => ReferenceExpression.Create($"{DefaultUserName}");
+
+    private static ReferenceExpression PasswordReference => ReferenceExpression.Create($"{DefaultPassword}");
+
+    /// <summary>
+    /// ConnectionString for the LavinMQ server in the form of amqp://guest:guest@host:port/.
+    /// </summary>
+    public ReferenceExpression ConnectionStringExpression =>
+        LavinMQConnectionString.Create(UserNameReference, PasswordReference, PrimaryEndpoint);
```

Comment:
Is there a reason you introduced a class rather than using `ReferenceExpression.Create` like other integrations?

Reply:
Ahh. This was purely because Lavin could introduce env var config. If they did, we’d have to switch on VirtualHost being null or not to create the connection string with optional expressions. 

Right now they dont support that though, so I could change it?